src/components/MasterItems.js

```diff
@@ -0,0 +1,69 @@
+import { Component, PropTypes, DOM as dom, createFactory } from 'react';
+import MasterItemsSelectComponent from './MasterItemsSelect';
+import _ from 'underscore';
+
+const MasterItemsSelect = createFactory(MasterItemsSelectComponent);
+
+class MasterItems extends Component {
+  constructor(props) {
+    super(props);
+    this.selectItem = this.selectItem.bind(this);
+    this.addComment = this.addComment.bind(this);
+    this.handleAddEntry = this.handleAddEntry.bind(this);
+    this.state = {
+      comment: ''
+    };
+  }
+
+  handleAddEntry() {
+    const selectedItem = _.extend({}, this.state.selectedItem || this.props.masterItems[0]);
+    selectedItem.comment = this.state.comment;
+    const entries = this.props.entries.concat(selectedItem);
+
+    this.setState({ comment: '' });
+    this.props.handleUpdateEntries(entries);
+  }
+
+  selectItem(itemId) {
+    const selectedItem = _.extend({}, _.find(this.props.masterItems, (item) => item.id === itemId));
+
+    this.setState({ selectedItem });
+  }
+
+  addComment(e) {
+    const comment = e.target.value;
+
+    this.setState({ comment });
+  }
+
+  render() {
+    return dom.div(
+      null,
+      dom.h1(null, 'New Order'),
```

Comment:
Should this still be part of the `NewOrder` component? Or at least use a `prop` so we can reuse for adding entries to an existing order, like you mentioned?